stdlib/LinearAlgebra/test/diagonal.jl

```diff
@@ -726,4 +726,12 @@ end
     @test dot(zeros(Int32, 0), Diagonal(zeros(Int, 0)), zeros(Int16, 0)) === 0
 end
 
+# issue 37359
+@testset "Issue #37359 return Diagonal Matrix for Abstract type :> Number" begin
+    D = Diagonal(rand(Int, n))
+    D2 = Diagonal{Integer}(rand(Int, n))
+    @test isa(I \ Diagonal(ones(Integer, n)), Diagonal)
+    @test D \ D2 ≈ Diagonal(Matrix(D) \ Matrix(D2))
```

Comment:
Doesn't this test another method? I'd imagine something like
```julia
v = Integer[-5:5...]
@test 2I \ Diagonal(v) == 2\Diagonal(v) == Diagonal(v/2)
```

Reply:
Added the test you've mentioned above, let me know if you want me to remove the one I added for computation invoking the matrices

Reply:
:man_shrugging: I thinks it's basically a duplicate of https://github.com/JuliaLang/julia/blob/60c36f39cca453b6875ce8f530cfe0d5657daeae/stdlib/LinearAlgebra/test/diagonal.jl#L192
OTOH, it should not hurt to test this once more, as the time needed is negligible.

Reply:
Yes but I think the types of D and D2 were being derived from the below line which didn't include Abstract Types relevant for this issue?
```
@testset for relty in (Float32, Float64, BigFloat), elty in (relty, Complex{relty})
```
Please let me know if any other changes are required, happy to learn and improve